packages/astro/src/@types/astro.ts

```diff
@@ -2184,6 +2185,258 @@ export interface AstroUserConfig {
 		 * For a complete overview, and to give feedback on this experimental API, see the [Server Islands RFC](https://github.com/withastro/roadmap/pull/963).
 		 */
 		serverIslands?: boolean;
+
+		/**
+		 * @docs
```

Comment:
OK, to make things simpler, I have mocked up something in a branch of docs itself with an idea of what I'm thinking: https://content-layer-draft--astro-docs-2.netlify.app/en/reference/configuration-reference/#experimentalcontentlayer

I did take take a little from the RFC and mix it in here. I mostly tried to laser-focus things a bit because it's already quite long, and for *experimental* docs, we can be a bit more minimal and let the RFC do some of the work. 

Putting on my "seeing this for the first time, want to play with it" hat, I feel like it's the `loader` stuff that people need to get is fundamentally different, and most important, in all this. So I spent a bit more time there, and didn't spend as much time on the querying and rendering (even combined them into one section!) because they work so much like content collections do already.

So see what you think about something that kind of takes this form! When we have a good shape, then we can squish it in here to this file!

Reply:
That's great. Is it on a branch somewhere? Also do you have any suggestions for a workflow for working on this? It's really hard editing inside the JSDoc!